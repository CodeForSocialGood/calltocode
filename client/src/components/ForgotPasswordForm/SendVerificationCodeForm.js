import React from 'react'
import {Field, reduxForm} from 'redux-form'
import styles from './ForgotPasswordForm.scss'
import PropTypes from 'prop-types'

function EmailField ({ input }) {
  return (
    <input
      className={styles.inputEmail}
      placeholder="Email"
      {...input} />
  )
}

const SendVerificationCodeForm = props => {
  const {handleSubmit} = props
  return (

    <form onSubmit={handleSubmit}
      className={styles.form}>

      <h1
        className={styles.title}>
            Forgot Password?
      </h1>

      <h3>{"Let's get you a new one!"}</h3>

      <Field
        name="email"
        component={EmailField} />

      <button
        className={styles.buttonSubmit}
        type="submit">
            Send Verification Code
      </button>
    </form>
  )
}

EmailField.propTypes = {
  input: PropTypes.object
}
SendVerificationCodeForm.propTypes = {
  handleSubmit: PropTypes.func
}

export default reduxForm({
  form: 'ForgotPasswordForm', // <------ same form name
  destroyOnUnmount: false, // <------ preserve form data
  forceUnregisterOnUnmount: true // <------ unregister fields on unmount
})(SendVerificationCodeForm)
