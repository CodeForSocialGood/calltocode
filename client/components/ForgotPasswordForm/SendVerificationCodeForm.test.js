/* eslint-env jest */
import React from 'react'
import {shallow} from 'enzyme'

import {SendVerificationCodeForm} from './SendVerificationCodeForm'
let mockProps

beforeEach(() => {
  mockProps = {
    email: 'fakeEmail',
    onChangeEmail: () => {},
    handleSubmit: () => {}
  }
})

describe('<SendVerificationCodeForm />', () => {
  describe('email is length 0', () => {
    beforeEach(() => { mockProps.email = '' })

    test('component renders correctly', () => {
      const component = shallow(<SendVerificationCodeForm {...mockProps} />)
      expect(component).toMatchSnapshot()
    })
  })

  describe('email is not length 0', () => {
    beforeEach(() => { mockProps.email = 'some@email.com' })

    test('component renders correctly', () => {
      const component = shallow(<SendVerificationCodeForm {...mockProps} />)
      expect(component).toMatchSnapshot()
    })
  })
})
