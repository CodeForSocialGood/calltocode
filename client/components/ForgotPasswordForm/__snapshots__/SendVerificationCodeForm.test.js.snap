// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SendVerificationCodeForm /> email is length 0 component renders correctly 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <h1
    className="title"
  >
    Forgot Password?
  </h1>
  <h3>
    Let's get you a new one!
  </h3>
  <Field
    component={[Function]}
    name="email"
    onChange={[Function]}
    value=""
  />
  <button
    className="buttonSubmit"
    disabled={true}
    type="submit"
  >
    Send Verification Code
  </button>
</form>
`;

exports[`<SendVerificationCodeForm /> email is not length 0 component renders correctly 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <h1
    className="title"
  >
    Forgot Password?
  </h1>
  <h3>
    Let's get you a new one!
  </h3>
  <Field
    component={[Function]}
    name="email"
    onChange={[Function]}
    value="some@email.com"
  />
  <button
    className="buttonSubmit"
    disabled={false}
    type="submit"
  >
    Send Verification Code
  </button>
</form>
`;
